clients/roscpp/include/ros/timer_manager.h

```diff
@@ -584,7 +584,7 @@ void TimerManager<T, D, E>::threadFunc()
       {
         // On system time we can simply sleep for the rest of the wait time, since anything else requiring processing will
         // signal the condition variable
-        int32_t remaining_time = std::max((int32_t)((sleep_end - current).toSec() * 1000.0f), 1);
+        int64_t remaining_time = std::max((int64_t)((sleep_end - current).toSec() * 1000.0f), (int64_t)1);
```

Comment:
IMO the second cast should not be required in order for `std::max` to do the right thing.

Can you also use `static_cast` vs a c-style cast, please?

Reply:
> IMO the second cast should not be required in order for std::max to do the right thing.

Gave me the following error in that case: `deduced conflicting types for parameter ‘const _Tp’ (‘long int’ and ‘int’)`. Explicitly templating `std::max<int64_t>(...)` would probably have been better than casting both arguments. With my latest commit it's a `static_cast<int64_t>(std_max<double>(...))`

Reply:
Is there something which prevents doing `std::max<int64_t>`? That does indeed seem better than doing the `max` against a floating point value that's then cast back to an integer.